.github/integration/tests/sda/60_api_admin_test.sh

```diff
@@ -104,27 +101,18 @@ fi
 
 # Re-upload the file and use the api to ingest it, then try to delete it
 s3cmd -c s3cfg put NA12878.bam.c4gh s3://test_dummy.org/NE12878.bam.c4gh
+echo "waiting for upload to complete"
+sleep 10
 
-URI=http://rabbitmq:15672
-if [ -n "$PGSSLCERT" ]; then
-   URI=https://rabbitmq:15671
+if [ $((stream_size++)) -eq "$(curl -s -u guest:guest $URI/api/queues/sda/inbox | jq '.messages_ready')" ]; then
+   echo "Upload did not complete successfully"
+   exit 1
```

Comment:
If it randomly takes more than `sleep 10`, will this fail randomly as well?

Reply:
Reworked in 3353a1f